### PR TITLE
Parametrized stuff from general.h

### PR DESCRIPTION
**Pull request by lobradov**
Guard a lot of parameters with #ifndefs.

### CODE REVIEW

File: code/espurna/config/general.h [hunk 1]
```diff
@@ -333,53 +373,86 @@ PROGMEM const char* const custom_reset_string[] = {
...
+#endif
+
+#ifndef WIFI_SCAN_NETWORKS
 #define WIFI_SCAN_NETWORKS      1                   // Perform a network scan before connecting
+#endif
 
 // Optional hardcoded configuration (up to 2 networks)
 #ifndef WIFI1_SSID
 #define WIFI1_SSID              ""
-#endif
```
Comment (mcspr) [inline code/espurna/config/general.h]: #endif (just a reminder, WIFI2 below missing too :)

File: code/espurna/config/general.h [hunk 2]
```diff
@@ -333,53 +373,86 @@ PROGMEM const char* const custom_reset_string[] = {
...
+
 #ifndef WIFI1_MASK
 #define WIFI1_MASK              ""
 #endif
+
 #ifndef WIFI1_DNS
 #define WIFI1_DNS               ""
 #endif
+
+#endif // WIFI1
```
Comment (mcspr) [inline code/espurna/config/general.h]: And this is where it got lost. If ssid is defined other values are undefined.
https://github.com/xoseperez/espurna/blob/fa20c47f2905c46628cbdde475e899ecc30368a3/code/espurna/wifi.ino#L174-L183

File: code/espurna/config/general.h [hunk 3]
```diff
@@ -839,14 +1076,32 @@ PROGMEM const char* const custom_reset_string[] = {
 #define NTP_SUPPORT             1               // Build with NTP support by default (6.78Kb)
 #endif
 
+#ifndef NTP_SERVER
 #define NTP_SERVER              "pool.ntp.org"  // Default NTP server
+#endif
+
+#ifndef NTP_TIMEOUT
```
Comment (mcspr) [inline code/espurna/config/general.h]: Besides timeout guards both NTP_TIME_OFFSET and NTP_DAY_LIGHT

File: code/espurna/config/general.h (hunk 2 above)
Comment (lobradov) [reply code/espurna/config/general.h]: well, it's kind of "by design"  - if you don't specify WIFI1 SSID, than rest of the WIFI1 settings are pointless. I can change back to "old" behavior, but I see no usecase of setting PASS or GW, without actually setting SSID.

File: code/espurna/config/general.h (hunk 3 above)
Comment (lobradov) [reply code/espurna/config/general.h]: done.

File: code/espurna/config/general.h [hunk 4]
```diff
@@ -395,35 +468,60 @@ PROGMEM const char* const custom_reset_string[] = {
 
 // This is not working at the moment!!
 // Requires ASYNC_TCP_SSL_ENABLED to 1 and ESP8266 Arduino Core 2.4.0
+#ifndef WEB_SSL_ENABLED
 #define WEB_SSL_ENABLED         0           // Use HTTPS web interface
+#endif
 
-#define WEB_MODE_NORMAL         0
-#define WEB_MODE_PASSWORD       1
```
Comment (mcspr) [inline code/espurna/config/general.h]: This is from #780

File: code/espurna/config/general.h (hunk 2 above)
Comment (mcspr) [reply code/espurna/config/general.h]: I am mostly talking about logic above. If WIFI1_SSID is defined, but nothing else is:
```
espurna/code/espurna/wifi.ino: In function 'void _wifiInject()':
espurna/code/espurna/wifi.ino:178:35: error: 'WIFI1_PASS' was not declared in this scope
setSetting("pass", 0, WIFI1_PASS);
^
espurna/code/espurna/wifi.ino:179:33: error: 'WIFI1_IP' was not declared in this scope
setSetting("ip", 0, WIFI1_IP);
^
espurna/code/espurna/wifi.ino:180:33: error: 'WIFI1_GW' was not declared in this scope
setSetting("gw", 0, WIFI1_GW);
^
code/espurna/wifi.ino:181:35: error: 'WIFI1_MASK' was not declared in this scope
setSetting("mask", 0, WIFI1_MASK);
^
code/espurna/wifi.ino:182:34: error: 'WIFI1_DNS' was not declared in this scope
setSetting("dns", 0, WIFI1_DNS);
```

Comment (lobradov) [reply code/espurna/config/general.h]: erm, right...
didn't see that in Travis (still shows success even if build fails).

**Pull request merged**